src/lib/axes.typ

```diff
@@ -128,9 +130,10 @@
   let l = ()
   if ticks != none {
     let major-tick-values = ()
-    if "step" in ticks and ticks.step != none {
+    if "step" in ticks and ticks.step != none and ticks.step != 0 {
       assert(ticks.step >= 0,
              message: "Axis tick step must be positive")
```

Comment:
Why not panic if `ticks.step` is 0? Also to check use `ticks.step not in (none, 0)`